tensorflow/core/ops/array_ops.cc

```diff
@@ -1368,8 +1368,12 @@ REGISTER_OP("UniqueV2")
     .Attr("Taxis: {int32,int64} = DT_INT64")
     .Attr("out_idx: {int32, int64} = DT_INT32")
     .SetShapeFn([](InferenceContext* c) {
-      c->set_output(0, c->Vector(InferenceContext::kUnknownDim));
-      c->set_output(1, c->input(0));
+      int32 rank = InferenceContext::Rank(c->input(0));
+      std::vector<DimensionHandle> dims;
+      dims.assign(rank, c->MakeDim(InferenceContext::kUnknownDim));
```

Comment:
Since `InferenceContext::Rank` can return `InferenceContext::kUnknownRank` (which is -1), we should check if rank is nonnegative before assigning values to `dims`.

Reply:
What should we do if the shape of input(0) is not set? Return an error with message? Or just ignore it?

Reply:
I'm thinking either returning an `errors::InvalidArgument` or just use a `DCHECK` (but `Rank()` actually already has a `DCHECK` for that). 

Let's ask @rmlarsen. What do you recommend, Rasmus? Thank you!

Reply:
It seems @rmlarsen is overloaded. @ezhulenev, could you please help answer?

Reply:
Unknown rank seems to be totally valid input, and unknown input rank must produce unknown output rank.

I spent some time reading through the https://cc.corp.google.com/piper///depot/google3/third_party/tensorflow/core/kernels/unique_op.cc?type=cs&q=UniqueV2&sq=package:piper+file://depot/google3+-file:google3/experimental&g=0&l=53 implementation, it would really help if you can add a bunch of test cases for different input ranks/dims and expected output ranks/dims. Right now it's really hard to tell if it's correct or not without diving deep into unique_op implementation.